MonoGame.Framework.Content.Pipeline/Graphics/Font/CharacterRegionTypeConverter.cs

```diff
@@ -4,68 +4,93 @@
+        {
+            // Input must be a string.
+            string source = value as string;
 
-			if (string.IsNullOrEmpty(source))
-			{
-				throw new ArgumentException();
-			}
+            if (string.IsNullOrEmpty(source))
+            {
+                throw new ArgumentException("Input string cannot be null or empty.");
+            }
 
-			// Supported input formats:
-			//  A
-			//  A-Z
-			//  32-127
-			//  0x20-0x7F
+            // Parse as single character or range (e.g., "A", "A-Z", "32-127", "0x20-0x7F").
+            var splitStr = source.Split('-');
+            var split = new char[splitStr.Length];
+            for (int i = 0; i < splitStr.Length; i++)
+            {
+                split[i] = ConvertCharacter(splitStr[i]);
+            }
 
-			var splitStr = source.Split('-');
-			var split = new char[splitStr.Length];
-			for (int i = 0; i < splitStr.Length; i++)
-			{
-				split[i] = ConvertCharacter(splitStr[i]);
-			}
+            switch (split.Length)
+            {
+                case 1:
+                    // Only a single character (eg. "a").
+                    return new CharacterRegion(split[0], split[0]);
 
-			switch (split.Length)
-			{
-				case 1:
-				// Only a single character (eg. "a").
-				return new CharacterRegion(split[0], split[0]);
+                case 2:
+                    // Range of characters (eg. "a-z").
+                    return new CharacterRegion(split[0], split[1]);
 
-				case 2:
-				// Range of characters (eg. "a-z").
-				return new CharacterRegion(split[0], split[1]);
+                default:
+                    throw new ArgumentException("Input format is not supported.");
+            }
+        }
 
-				default:
-				throw new ArgumentException();
-			}
-		}
 
-
-		static char ConvertCharacter(string value)
-		{
-			if (value.Length == 1)
-			{
-				// Single character directly specifies a codepoint.
-				return value[0];
-			}
-			else
-			{
-				// Otherwise it must be an integer (eg. "32" or "0x20").
-				return (char)(int)intConverter.ConvertFromInvariantString(value);
-			}
-		}
-
-
-		static TypeConverter intConverter = TypeDescriptor.GetConverter(typeof(int));
-	}
-}
\ No newline at end of file
+        static char ConvertCharacter(string value)
+        {
+            if (value.Length == 1)
+            {
+                // Single character directly specifies a codepoint.
+                return value[0];
+            }
+            else
+            {
+                // Check if the value is hexadecimal with the "0x" prefix
+                if (value.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Remove "0x" prefix and parse as hexadecimal
+                    value = value.Substring(2);
+                    if (int.TryParse(value, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out int hexResult))
```

Comment:
I would change `value` here to `value.Substring(2)`.  This keeps you from changing the value of `value` so if this fails the exception message below gets the original value of it.